extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/internal/job/EventJob.java

```diff
@@ -7,26 +7,89 @@
  */
 package org.eclipse.smarthome.binding.astro.internal.job;
 
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.eclipse.smarthome.binding.astro.internal.job.Job.checkNull;
+
 import org.eclipse.smarthome.binding.astro.handler.AstroThingHandler;
 import org.eclipse.smarthome.binding.astro.internal.AstroHandlerFactory;
-import org.quartz.JobDataMap;
 
 /**
- * Job to trigger a event.
+ * Scheduled Job to trigger Events
  *
  * @author Gerhard Riegler - Initial contribution
+ * @author Amit Kumar Mondal - Implementation to be compliant with ESH Scheduler
  */
-public class EventJob extends AbstractBaseJob {
-    public static final String KEY_EVENT = "event";
+public final class EventJob implements Job {
+
+    private final String thingUID;
+    private final String channelID;
+    private final String event;
 
+    /**
+     * Constructor
+     *
+     * @param thingUID
+     *            thing UID
+     * @param channelID
+     *            channel ID
+     * @param event
+     *            Event name
+     * @throws NullPointerException
+     *             if any of the arguments is {@code null}
+     */
+    public EventJob(String thingUID, String channelID, String event) {
+        checkNotNull(thingUID, "Thing UID cannot be null");
+        checkNotNull(channelID, "Channel ID cannot be null");
+        checkNotNull(event, "Event cannot be null");
```

Comment:
Please change it to a normal `null` check and throw a `IllegalArgumentException`. This is more meaningful than throwing an NPE here.